Microsoft.Toolkit.Uwp.UI.Controls.DataGrid/DataGrid/DataGridComboBoxColumn.cs

```diff
@@ -0,0 +1,606 @@
+            {
+                RefreshForeground(textBlock, computedRowForeground);
+            }
+        }
+
+        private void RefreshForeground(ComboBox comboBox, Brush computedRowForeground)
+        {
+            if (this.Foreground == null)
+            {
+                if (computedRowForeground != null)
+                {
+                    comboBox.Foreground = computedRowForeground;
+                }
+            }
+            else
+            {
+                comboBox.Foreground = this.Foreground;
+            }
+        }
+
+        private void RefreshForeground(TextBlock textBlock, Brush computedRowForeground)
+        {
+            if (this.Foreground == null)
+            {
+                if (computedRowForeground != null)
+                {
+                    textBlock.Foreground = computedRowForeground;
+                }
+            }
+            else
+            {
+                textBlock.Foreground = this.Foreground;
+            }
+        }
+
+        private void SetTextFontSize(DependencyObject textElement, DependencyProperty fontSizeProperty)
+        {
+            double newFontSize = this.FontSize;
+            if (double.IsNaN(newFontSize))
+            {
+                textElement.ClearValue(fontSizeProperty);
+            }
+            else
+            {
+                textElement.SetValue(fontSizeProperty, newFontSize);
+            }
+        }
+
+        private bool EnsureOwningGrid()
+        {
+            if (this.OwningGrid != null)
+            {
+                if (this.OwningGrid != _owningGrid)
+                {
+                    _owningGrid = this.OwningGrid;
+                    _owningGrid.LoadingRow += OwningGrid_LoadingRow;
+                    _owningGrid.CellEditEnded += OwningGrid_CellEditEnded;
+                }
+
+                return true;
+            }
+
+            return false;
+        }
+
+        private void OwningGrid_LoadingRow(object sender, DataGridRowEventArgs e)
+        {
+            SetDisplayMemberPathValue(e.Row);
+        }
+
+        private void OwningGrid_CellEditEnded(object sender, DataGridCellEditEndedEventArgs e)
+        {
+            SetDisplayMemberPathValue(e.Row);
+        }
+
+        private void SetDisplayMemberPathValue(DataGridRow row)
+        {
+            if (this.OwningGrid != null)
+            {
+                var textBlock = this.GetCellContent(row) as TextBlock;
+                if (textBlock != null && DisplayMemberPath != null)
+                {
+                    var displayValue = GetDisplayValue(row.DataContext);
+
+                    textBlock.Text = displayValue;
+                }
+            }
+        }
+
+        private string GetDisplayValue(object dataItem)
+        {
+            if (this.Binding != null && dataItem != null)
+            {
+                var value = dataItem.GetType().GetProperty(Binding?.Path.Path).GetValue(dataItem);
```

Comment:
Thanks Filip.   Binding?.Path.Path can now be Binding.Path.Path

Reply:
By the way, I would use
   if (this.Binding != null && this.Binding.Path != null && dataItem != null) 
because this.Binding.Path alone can be null.

Reply:
Agreed